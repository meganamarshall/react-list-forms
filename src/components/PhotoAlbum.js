import React from 'react';
import PropTypes from 'prop-types';
import Photos from './Photos';

function PhotoAlbum({ photos, title }) {
  return (
    <section>
      <h2>{title}</h2>
      <Photos photos={photos} />
    </section>
  );
}

PhotoAlbum.propTypes = {
  photos: PropTypes.array.isRequired,
  title: PropTypes.string.isRequired
};

export default PhotoAlbum;
