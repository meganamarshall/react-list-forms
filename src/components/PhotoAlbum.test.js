import React from 'react';
import { shallow } from 'enzyme';
import PhotoAlbum from './PhotoAlbum';
import candyByColor from '../assets/candyByColor.jpg';
import gettyImages from '../assets/gettyimages-649680072.jpg';
import images from '../assets/images.jpeg';
import othercandy from '../assets/othercandy.jpg';

describe('photoalbum component', () => {
  it('renders the photo album', () => {
    const title = 'Hello';
    const photos = [
      candyByColor,
      gettyImages,
      images,
      othercandy
    ];
    const wrapper = shallow(<PhotoAlbum title={title} photos={photos} />);
    expect(wrapper).toMatchSnapshot();
  });
});
