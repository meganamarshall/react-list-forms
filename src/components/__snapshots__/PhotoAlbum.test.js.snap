// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`photoalbum component renders the photo album 1`] = `
<section>
  <h2>
    Hello
  </h2>
  <Photos
    photos={
      Array [
        "/path/image",
        "/path/image",
        "/path/image",
        "/path/image",
      ]
    }
  />
</section>
`;
